### ISSUE TITLE
where to connect buttons?

### ISSUE THREAD

**Issue by drschnalli** [WLAN-Pi/feedback#16]
Hey there i dont exactly know in which forum or so i have to ask.. but does someone know where i can connect the buttons to? So that navigation via oled will work?

**Comment by ThatIdiotSid** [comment]
> Hey there i dont exactly know in which forum or so i have to ask.. but does someone know where i can connect the buttons to? So that navigation via oled will work?

That Waveshare 128x128, 1.44inch LCD display HAT?    It just slotted into the Pi 4B  GPIO exposed with the Geekworm case I used.  (insert the hat,   screwed two sides down,  and used non-conductive adhesive so the other side of that hat laid flat and didn't contact the case).

In the 3.0.0 release,  I couldn't get the OLED HAT recognized, so I went to the vendors website and tried installing various dependencies, only to notice they'd omitted a *really important one part of it*,  so it didn't work.  That was resolved in the PiOS 3.0.1 release, and those three keys and that switch on the left work just fine now.

**Comment by jiribrejcha** [comment]
Here are the GPIO numbers for Waveshare HAT buttons used by the [Raspberry Pi 4 based Community Edition](https://userguide.wlanpi.com/hardware/wlan-pi-community-edition/based-on-raspberry-pi-4):

```
# Button mapping for the Waveshare 1.44 inch LCD Display HAT
# https://www.waveshare.com/1.44inch-lcd-hat.htm
BUTTONS_WAVESHARE = {
    "up": 6,
    "down": 19,
    "left": 5,
    "right": 26,
    "center": 13,
    "key1": 21,
    "key2": 20,
    "key3": 16
}
```

GPIO info for all WLAN Pi models is [here, line 101 onwards](https://github.com/WLAN-Pi/wlanpi-fpms/blob/main/fpms/modules/constants.py).
